### ISSUE TITLE
Missing ‘type’ or ‘collectiondate’ error in logs

### ISSUE THREAD

**Issue by nickfennell** [robbrad/UKBinCollectionData#1073]
### Name of Council

Rugby Borough Council

### Issue Information

Error returned is below.

`This error originated from a custom integration.

Logger: custom_components.uk_bin_collection
Source: custom_components/uk_bin_collection/__init__.py:282
integration: UK Bin Collection Data (documentation, issues)
First occurred: 01:19:27 (12 occurrences)
Last logged: 01:33:47

[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '12/12/2024'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '19/12/2024'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '28/12/2024'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '02/01/2025'}
`

### Verification

- [X] I searched for similar issues at https://github.com/robbrad/UKBinCollectionData/issues?q=is:issue and found no duplicates
- [X] I have checked my address/postcode/UPRN works on the council's website
- [X] I have provided a detailed explanation of the issue as well as steps to replicate the issue
- [X] I understand that this project is run by volunteer contributors therefore completion of this issue cannot be guaranteed

**Comment by m26dvd** [comment]
@nickfennell Is this still a problem? Bin types return okay for me.

If you are still having issues, can you provide an example UPRN/Postcode for testing purposes please?

**Comment by dp247** [comment]
Yeah, they're fine for me too. If you don't want to put your info here, you can find my email on my Github profile.

**Comment by nickfennell** [comment]
Hi.

Apologies for slow reply. Updated the integration to 0.123.1 today.

Error shown in log file is as below:

```

This error originated from a custom integration.

Logger: custom_components.uk_bin_collection
Source: custom_components/uk_bin_collection/__init__.py:282
integration: UK Bin Collection Data (documentation, issues)
First occurred: 23:12:38 (4 occurrences)
Last logged: 23:12:38

[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '19/12/2024'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '28/12/2024'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '02/01/2025'}
[UKBinCollection] Missing 'type' or 'collectionDate' in bin data: {'type': None, 'collectionDate': '09/01/2025'}
```

**Collection URL**: https://www.rugby.gov.uk/check-your-next-bin-day
**URN is**: 10094755105
**Postcode is**: CV23 1FH

Thanks!

**Comment by dp247** [comment]
When I originally wrote this parser, I decided to map bins to colours. The issue was that none of the bin mappings could be found for the provided type, so it was throwing nulls. I've changed it to just pull the bin type and make it pretty now.

**Comment by nickfennell** [comment]
Understood.

Will you be pushing a commit soon?

Thank you for investigating.

**Comment by dp247** [comment]
Yeah, it's fixed in #1088
